upstream-sha256: 59bd76d5c647f3b1f93740ab45a42cd55ac9738fb06dc6e263a5e36759fcf9b0
actual-sha256: b05e334552b534ff5cc3454aee8e375ea76cf55c4dd5375be9fd7a75ac2f6764
--- upstream.expected
+++ tolk.actual
@@ -14,7 +14,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((__global float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3));
+  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
 }
 ---
 __kernel void r_4_4_4_4(__global float* data0_16, __global float* data1_16, __global float* data2_16, __global float* data3_16, __global float* data4_16) {
@@ -33,7 +33,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((__global float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = (2.0f*((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
+  *(data0_16+(lidx1+alu0)) = (2.0f*((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3))));
 }
 ---
 __kernel void r_4_4_4(__global float* data0_16, __global float* data1_16, __global float* data2_16, __global float* data3_16) {
@@ -44,7 +44,7 @@
   float val0 = (*(data1_16+alu1));
   float4 val1 = (*((__global float4*)((data2_16+alu0))));
   float4 val2 = (*((__global float4*)((data3_16+(lidx1<<2)))));
-  *(data0_16+alu1) = ((2.0f*val0)+(val1.x*val2.x)+(val1.y*val2.y)+(val1.z*val2.z)+(val1.w*val2.w));
+  *(data0_16+alu1) = ((2.0f*val0)+(0.0f+(val1.x*val2.x)+(val1.y*val2.y)+(val1.z*val2.z)+(val1.w*val2.w)));
 }
 ---
 __kernel void r_4_4_4_4(__global float* data0_16, __global float* data1_16, __global float* data2_16, __global float* data3_16, __global float* data4_16) {
@@ -66,7 +66,7 @@
   float val7 = (*(data4_16+(lidx1+12)));
   float val8 = (*(data4_16+(lidx1+4)));
   float val9 = (*(data4_16+(lidx1+8)));
-  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7));
+  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(0.0f+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7)));
 }
 ---
 __kernel void r_4_4_4_4(__global float* data0_16, __global float* data1_16, __global float* data2_16, __global float* data3_16, __global float* data4_16) {
@@ -88,7 +88,7 @@
   float val7 = (*(data4_16+(lidx1+12)));
   float val8 = (*(data4_16+(lidx1+4)));
   float val9 = (*(data4_16+(lidx1+8)));
-  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7));
+  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(0.0f+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7)));
 }
 ---
 __kernel void r_4_4_4(__global float* data0_16, __global float* data1_16, __global float* data2_16) {
@@ -97,5 +97,5 @@
   float4 val0 = (*((__global float4*)((data1_16+alu0))));
   int lidx1 = get_local_id(1); /* 4 */
   float4 val1 = (*((__global float4*)((data2_16+(lidx1<<2)))));
-  *(data0_16+(lidx1+alu0)) = ((val0.x*val1.x)+(val0.y*val1.y)+(val0.z*val1.z)+(val0.w*val1.w));
+  *(data0_16+(lidx1+alu0)) = (0.0f+(val0.x*val1.x)+(val0.y*val1.y)+(val0.z*val1.z)+(val0.w*val1.w));
 }
